python/pyarrow/parquet.py

```diff
@@ -35,8 +35,9 @@
                               FileMetaData, RowGroupMetaData,
                               ColumnChunkMetaData,
                               ParquetSchema, ColumnSchema)
-from pyarrow.filesystem import (LocalFileSystem, _ensure_filesystem,
-                                resolve_filesystem_and_path)
+from pyarrow.filesystem import (
```

Comment:
These are legacy filesystem imports? Do we still need them?

Reply:
Perhaps `import pyarrow.filesystem as legacyfs` would make the code easier to read below.

Reply:
Yes, we still need them because the full ParquetDataset/ParquetManifest (python) implementation here is based on the legacy filesystems. 
But switched to use `legacyfs.` for the old ones, and plain imports for the new ones